frontend/src/app/apiSlice.ts

```diff
@@ -0,0 +1,36 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+function getApiUrl(): string {
+  // Avoid an obvious poor environment setup footgun
+  if (process.env.EXPO_PUBLIC_API_DOMAIN == null) {
+    throw new Error("API URL not set");
+  }
+
+  // http in development, https in production
+  if (process.env.NODE_ENV === "development") {
+    return "http://" + process.env.EXPO_PUBLIC_API_DOMAIN;
```

Comment:
why is dev api still expo_public?

Reply:
The same variable is used for the domain in prod and in dev, so in production only the actual API domain should be bundled in.